crates/lib/src/token/tests.rs

```diff
@@ -80,4 +183,312 @@ mod tests {
+        assert_eq!(ix.accounts[0].pubkey, source);
+        assert_eq!(ix.accounts[1].pubkey, mint);
+        assert_eq!(ix.accounts[2].pubkey, dest);
+        assert_eq!(ix.accounts[3].pubkey, authority);
+    }
+
+    #[test]
+    fn test_token2022_ata_derivation() {
+        let program = TokenProgram::new(TokenType::Token2022);
+        let wallet = Pubkey::new_unique();
+        let mint = Pubkey::new_unique();
+
+        let ata = program.get_associated_token_address(&wallet, &mint);
+        
+        // Verify ATA derivation matches spl-token-2022
+        let expected_ata = spl_associated_token_account::get_associated_token_address_with_program_id(
+            &wallet,
+            &mint,
+            &spl_token_2022::id(),
+        );
+        assert_eq!(ata, expected_ata);
+    }
+
+    #[test]
+    fn test_token2022_transfer_fee_calculation() {
+        let mint = Pubkey::new_unique();
+        let owner = Pubkey::new_unique();
+        let amount = 1_000_000;
+
+        // Test with 1% fee (100 basis points)
+        let transfer_fee = TransferFeeConfig {
+            epoch: 0,
+            maximum_fee: 0,
+            transfer_fee_basis_points: 0,
+            newer_transfer_fee: TransferFeeAmount {
+                epoch: 1,
+                transfer_fee_basis_points: 100, // 1%
+                maximum_fee: 10_000,
+            },
+            older_transfer_fee: TransferFeeAmount {
+                epoch: 0,
+                transfer_fee_basis_points: 0,
+                maximum_fee: 0,
+            },
+        };
+
+        let account = Token2022Account {
+            mint,
+            owner,
+            amount,
+            transfer_fee: Some(transfer_fee),
+        };
+
+        // Test basic fee calculation
+        let fee = std::cmp::min(
+            (amount as u128 * 100u128 / 10000) as u64,
+            10_000,
+        );
+        assert_eq!(fee, 10_000);
+
+        // Test with smaller amount
+        let small_amount = 1_000;
+        let small_fee = std::cmp::min(
+            (small_amount as u128 * 100u128 / 10000) as u64,
+            10_000,
+        );
+        assert_eq!(small_fee, 10);
+    }
+
+    #[test]
+    fn test_token2022_account_state_extensions() {
+        let mint = Pubkey::new_unique();
+        let owner = Pubkey::new_unique();
+        let amount = 1000;
+
+        // Create account data with transfer fee extension
+        let mut buffer = vec![0; 1000];
+        let mut account = StateWithExtensions::<spl_token_2022::state::Account>::unpack_from_slice(&buffer).unwrap();
+        account.base.mint = mint;
+        account.base.owner = owner;
+        account.base.amount = amount;
+
+        // Pack the account back
+        account.pack_into_slice(&mut buffer);
+
+        // Test unpacking with our implementation
+        let program = TokenProgram::new(TokenType::Token2022);
+        let unpacked = program.unpack_token_account(&buffer).unwrap();
```

Comment:
Please add a test that takes a real token deployed to Solana (like PYUSD) and unpack it using this function and verify its details. You should hard code the token bytes in the test so it does not need a network connection